src/bsoncxx/document/view.cpp

```diff
@@ -124,7 +125,8 @@ view::const_iterator view::const_iterator::operator++(int) {
 }
 
 bool operator==(const view::const_iterator& lhs, const view::const_iterator& rhs) {
-    return (lhs._element.raw == rhs._element.raw && lhs._element.offset == rhs._element.offset);
+    return (lhs._element.raw() == rhs._element.raw() &&
```

Comment:
Sam helped me figure it out: std::tie requires its arguments to be l-values. Is there is a way to convert to l-values in one line? If not, performing two equality checks is probably cleaner.